holidays/registry.py

```diff
@@ -222,6 +225,11 @@ def __str__(self) -> str:
     def get_entity(self) -> Optional[HolidayBase]:
         """Return lazy-loaded entity."""
         if self.entity is None:
+            # Avoid deadlock due to importlib.import_module not being thread-safe by caching all
+            # the first imports in a dedicated thread.
+            ensure_futr = IMPORT_ENSURE_EXECUTOR.submit(importlib.import_module, self.module_name)
+            ensure_futr.result()  # We can discard the result, as it is now cached.
```

Comment:
This will be called on every import, but the lock is only held on the first import.